jsonnet/kube-prometheus/components/prometheus.libsonnet

```
@@ -65,12 +73,8 @@ function(params) {
     (import 'github.com/thanos-io/thanos/mixin/alerts/sidecar.libsonnet') +
     (import 'github.com/kubernetes-monitoring/kubernetes-mixin/lib/add-runbook-links.libsonnet') + {
       _config+:: p._config.mixin._config,
-      targetGroups: {},
-      sidecar: {
-        selector: p._config.mixin._config.thanosSelector,
-        thanosPrometheusCommonDimensions: 'namespace, pod',
-        dimensions: std.join(', ', ['job', 'instance']),
-      },
+      targetGroups+: p._config.mixin._config.thanos.targetGroups,
+      sidecar+: p._config.mixin._config.thanos.sidecar,
```

```suggestion
      # TODO: remove `_config.thanosSelector` after 0.10.0
      sidecar+: { selector: p._config.mixin._config.thanosSelector, } p._config.mixin._config.thanos.sidecar,
```

backwards compatibility layer